### ISSUE TITLE
[FoxyTab] Feature Request: Protect the tab

### ISSUE THREAD

**Issue by theogast** [erosman/support#261]
1. an option to protect / lock the tab would be welcome, with a marking of the tabs thus protected against accidental closing. tabmix (sniff ..) did it and it was very efficient
2. display site thumbnails when hovering over tabs?
3. close a tab by double-clicking on it
is this still possible on the new restrictive interface? euh.... new ?   yes !  I just switched to FF quantum because some sites now refuse to open with FF56 which I kept until today for extensions. I am therefore trying to redo a practical version of FF quantum ... hihi
in any case, thank you for your work which already helps us well
regards
![2020-11-27_175019](https://user-images.githubusercontent.com/75135084/100470723-113a4500-30d9-11eb-8572-31693449fb90.jpg)

**Comment by erosman** [comment]
> an option to protect / lock the tab would be welcome, with a marking of the tabs thus protected against accidental closing. tabmix (sniff ..) did it and it was very efficient

That can be annoying if there is pop-up every time user wants to close a tab. There is no way to know if a closure is accidental or deliberate.
Furthermore, pop-up can only be generated from page content of the tab, which doesn't work with some locations and requires interaction with page content that FoxyTab doest do ATM.

> display site thumbnails when hovering over tabs?

The thumbnail has to be available to be displayed which means getting it form 3rd-parties like Google etc. Firefox can capture but that would be a full size image and resource intensive for every tab.

> close a tab by double-clicking on it

Having to click twice instead of clicking once on the close ⨯, is twice the work.
Personally, I usually middle click on the tab to close it. ;)

**Comment by theogast** [comment]
thank you for your quick reply!
for the closing, I understand and I do like you, middle click. it's just a new habit to take
for the thumbnails, I still understand although I just found an extension that does it from the list of open tabs so suddenly, it suits my business (tabby)
on the other hand, I regret that you do not take into account the possibility of protecting certain tabs (is it impossible to do or very complicated?) because, when we have 100 open tabs and we want to close all those that do not are not used and therefore that we have not locked, it's really practical. Obviously to close those locked, you have to unlock them when you want to close them but at least impossible to close THE MIRACLE SITE THAT WE JUST FROM DISCOVER ;-) or the one we want to come back to later, if we protected it as soon as we saw that it was to be kept
in any case, thank you for your answer and for this constructive exchange
greetings
![2020-11-27_185854](https://user-images.githubusercontent.com/75135084/100476192-42b90d80-30e5-11eb-80da-1653c8c1ab41.jpg)

**Comment by erosman** [comment]
First, let's see if there is a popular demand for it. ;)

Tabby actually uses `tabs.captureTab()`. It is possible to do if there is a demand for it.

**Comment by theogast** [comment]
okay ! again thank you for your answer .... I would see indeed
........... HEY GUYS !  please come forward if you think this feature would be useful for you ... again I have always used it and it is very handy for managing its tabs open
friendships and thank you erosman

**Comment by erosman** [comment]
BTW, if you mean to PIN the tab, it is already in the next version of the FoxyTab.

**Comment by theogast** [comment]
yes I know that foxytab allows to pin the tabs but that does not prevent their accidental closing
this is not to pin the tab but to protect it against an involuntary closing, which is always possible when you have several dozen tabs open. thus those protected can only be closed after unlocking
it was very convenient with tabmix which did it

**Comment by philCryoport** [comment]
Hi @erosman -- as discussed in https://github.com/erosman/support/discussions/660 -- please add a feature to prevent a tab from being closed if toggled to prevent-closure or refresh:
```
* Toggle a Tab as "un-close-able" and "un-refreshable" via any of the following:
  * Clicking the X in the Tab
  * While viewing the Tab, hitting Ctrl-W (Windows & Linux) / Cmd-W (macOS)
  * Exiting the browser completely
  * Middle-clicking the Tab
  * Right-clicking a tab and clicking "Close Tab"
  * While viewing the Tab, clicking the refresh circular arrow
  * While viewing the Tab, hitting Ctrl-R or Ctrl-Shift-R (Windows & Linux) / Cmd-R or Cmd-Shift-R (macOS)
...somehow notifying the user that the Tab is toggled as "un-close-able" and "un-refreshable" when they do any of the above

...with the ability to toggle that "un-close-able" / "un-refreshable" Tab back to "close-able" / "refreshable" (i.e. normal behavior)
```

A few times now I've been in an Atlassian Chat Support session -- and accidentally closed the Tab -- and if I undo-close-tab (Cmd-Shift-T), the Chat Session is still lost :(
